include/ts/apidefs.h.in

```diff
@@ -873,6 +873,37 @@ typedef enum {
   TS_USER_ARGS_COUNT  ///< Fake enum, # of valid entries.
 } TSUserArgType;
 
+/** An enumeration of HTTP version types for the priority functions that behave
+ * differently across HTTP protocols. */
+typedef enum {
+  HTTP_PRIORITY_TYPE_HTTP_UNSPECIFIED = 1,
+  HTTP_PRIORITY_TYPE_HTTP_2,
+  HTTP_PRIORITY_TYPE_HTTP_3,
+} TSHttpPriorityType;
+
+/** The abstract type of the various HTTP priority implementations. */
+typedef struct {
+  /** The reference to the concrete HTTP priority implementation. This will be
+   * a value from TSHttpPriorityType. */
+  uint8_t priority_type;
+  /** The space allocated for the concrete priority implementation. */
+  uint8_t data[5];
```

Comment:
I would put this in `InkAPI.cc`
```
static_assert(sizeof(TSHttpPriority) >= sizeof(TSHttp2Priority));
```

Reply:
Good idea! And, as it turns out, padding made the size of TSHttp2Priority larger than TSHttpPriority.